ksql-serde/src/main/java/io/confluent/ksql/serde/json/JsonSerdeUtils.java

```diff
@@ -23,14 +23,66 @@
 import io.confluent.ksql.schema.connect.SchemaWalker.Visitor;
 import io.confluent.ksql.schema.ksql.PersistenceSchema;
 import io.confluent.ksql.schema.ksql.SqlBaseType;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.Schema.Type;
 
 final class JsonSerdeUtils {
 
+  // the JsonSchemaConverter adds a magic NULL byte and 4 bytes for the
+  // schema ID at the start of the message
+  private static final int SIZE_OF_SR_PREFIX = Byte.BYTES + Integer.BYTES;
+
   private JsonSerdeUtils() {
   }
 
+  /**
+   * Convert JSON serialized with {@link io.confluent.connect.json.JsonSchemaConverter}
+   * to standard JSON serialization.
+   *
+   * @param json the bytes representing the serialized JSON
+   * @return the serialized form with the magic byte and schemaID, or {@code json}
+   *         if it was not serialized with a magic byte to begin with
+   */
+  static byte[] removeMagicAndSchemaId(@Nullable final byte[] json) {
+    if (!hasMagicByte(json)) {
+      return json;
+    }
+
+    final byte[] out = new byte[json.length - SIZE_OF_SR_PREFIX];
+    System.arraycopy(json, SIZE_OF_SR_PREFIX, out, 0, json.length - SIZE_OF_SR_PREFIX);
+    return out;
+  }
```

Comment:
Seems to only be used in tests... is it needed?

Reply:
it was, now it's not, it will be again when we introduce `JSON_SR` format so I will keep it in for now 😂